base/loading.jl

```diff
@@ -364,7 +406,9 @@ function _require(mod::Symbol)
         if JLOptions().use_compilecache != 0
             doneprecompile = _require_search_from_serialized(mod, path)
             if !isa(doneprecompile, Bool)
-                return # success
+                M = doneprecompile[end]
```

Comment:
this list isn't sorted by that key (it's sorted by load order), I think you'll need to scan it to find when the module you wanted was defined